### PR TITLE
Add initial support for ARM based containers (AO-20418) 

### PR DESCRIPTION
**Pull request by d-maslyk**
Changes:
- align dockerfile to allow multi architecture builds - for amd64 and arm64
- yq install from github release
- add sanity check if given Snap Agent integration is supported

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,19 +7,25 @@ ARG DEBIAN_FRONTEND=noninteractive
...
-    apt-get -y upgrade && \
+    apt-get upgrade -y && \
     apt-get -y install \
       apt-transport-https \
       ca-certificates \
+      gnupg \
       docker.io \
-      curl \
-      yq && \ 
+      curl && \
```
Comment (lootek) [inline Dockerfile:16]: please arrange those alphabetically

File: Dockerfile [hunk 2]
```diff
@@ -7,19 +7,25 @@ ARG DEBIAN_FRONTEND=noninteractive
...
       apt-get clean && \
       rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+RUN arch="$(uname -m)" && if [ "${arch}" = "aarch64" ]; then \
+      yq_arch=arm64; \
+    elif [ "${arch}" = "x86_64" ]; then \
+      yq_arch=amd64; \ 
+    fi && \ 
+    curl -L "https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_${yq_arch}" -o yq && \
```
Comment (lootek) [inline Dockerfile:25]: yolo? why not from apt?

File: Dockerfile [hunk 3]
```diff
@@ -7,19 +7,25 @@ ARG DEBIAN_FRONTEND=noninteractive
...
+    mv yq /usr/bin/yq && \
+    chmod +x /usr/bin/yq
+
 ARG swisnap_version
 RUN echo "deb https://packagecloud.io/solarwinds/${swisnap_repo}/ubuntu/ bionic main" > /etc/apt/sources.list.d/swisnap.list && \
```
Comment (lootek) [inline Dockerfile:30]: bionic?

File: Dockerfile (hunk 2 above)
Comment (lootek) [reply Dockerfile:25]: no arm version

Comment (d-maslyk) [reply Dockerfile:25]: There is no arm yq version released by rmescandon in ppa:rmescandon APT repository.

File: Dockerfile (hunk 1 above)
Comment (d-maslyk) [reply Dockerfile]: Arranged these alphabetically.

File: Dockerfile (hunk 3 above)
Comment (d-maslyk) [reply Dockerfile]: Good catch. Fixed.

**Review by lootek — APPROVED** [review]

**Review by poleszcz — APPROVED** [review]

**Pull request merged**